runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/GaugeCell.java

```diff
@@ -50,6 +50,12 @@ public GaugeCell(MetricName name) {
     this.name = name;
   }
 
+  @Override
+  public void reset() {
+    dirty.afterModification();
```

Comment:
I don't believe reset() should make this dirty since reset() should set this to the state it is at as if it was newly constructed.

Reply:
The newly constructed DirtyState object is DIRTY by default. (https://github.com/apache/beam/blob/e51aa5f978c05f094310d67c4704017009121948/runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/DirtyState.java#L55). If it's set to CLEAN when reset, I think we should also change that behavior. However, I'm not sure whether this is by design and so I have not updated it for now. What's your thought?

Reply:
Your right, I was under the impression that it wasn't dirty initially so please ignore my comments about changing reset & dirty.